cdap-api/src/main/java/co/cask/cdap/api/schedule/DataSchedule.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.schedule;
+
+/**
+ *
+ */
+public class DataSchedule extends Schedule {
+  /**
+   * Source type, in case the {@link ScheduleType} of this schedule is {@code DATA}.
+   */
+  public enum SourceType { STREAM, DATASET }
```

Comment:
You're right we should have one ``NotificationSchedule``, and then have ``StreamSizeSchedule``, ``DatasetSizeSchedule``, ``DatasetNewFileSchedule`` classes extend from it - names could be different... 

This is how I see the whole thing working: there is only one scheduler service. It accepts any ``Schedule`` objects, and depending on the actual type of the a schedule, it uses an underlying scheduler that deals only with schedules of one type. So, there would be scheduler for ``TimeSchedule`` objects (that uses a quartz scheduler), one scheduler for the ``StreamSizeSchedule`` objects, one scheduler for the ``DatasetSizeSchedule`` objects, etc. Of course, the implementations of these objects could be only slightly different, so they could extend from one abstract scheduler.

A mix time/notification schedule could be just another ``Schedule``, probably extending from ``NotificationSchedule`` as well. 

What do you think of that design @anew ?